### ISSUE TITLE
[v3.9, iOS/Android] EditBox prints capital letters by default

### ISSUE THREAD

**Issue by rynkowsg** [cocos2d/cocos2d-x#15709]
problem: **EditBox by default uses capital letters.**
cocos2d-x version: **v3.9**
platform: **iOS, Android**

Code sample:

```
_emailEditBox = cocos2d::ui::EditBox::create( emailEditBoxBackground->getOriginalSize(),
                                              emailEditBoxBackground );
_emailEditBox->setInputFlag( cocos2d::ui::EditBox::InputFlag::SENSITIVE );
_emailEditBox->setInputMode( cocos2d::ui::EditBox::InputMode::EMAIL_ADDRESS );
_emailEditBox->setReturnType( cocos2d::ui::EditBox::KeyboardReturnType::DONE );
_emailEditBox->setFont( editBoxFontName.c_str(), editBoxFontSize );
_emailEditBox->setFontColor( BROWN );
_emailEditBox->setMaxLength( 50 );
_emailEditBox->setDelegate( this );
```

As a workaround, I set the input flag to SENSITIVE. On iOS this setting makes as a capital letter only for the first character. Unfortunately, **there is no flag to print all characters lowercase.**

![firstletteriscapital](https://cloud.githubusercontent.com/assets/5878299/15502090/c34a17b4-21a9-11e6-9bb3-a1c2e6b60b92.png)

On Android, EditBox with SENSITIVE flag shows all letters lowercase, but still SENSITIVE should be used for something else, not for avoiding capital letters.

**Comment by minggo** [comment]
@zilongshanren please take a look

**Comment by jasonchin** [comment]
Came across the same problem. The fix is in CCUIEditBoxIOS.mm:

changing lines 205-207 from

```
      case cocos2d::ui::EditBox::InputFlag::SENSITIVE:
            self.textInput.autocorrectionType = UITextAutocorrectionTypeNo;
            break;
```

to

```
      case cocos2d::ui::EditBox::InputFlag::SENSITIVE:
            self.textInput.autocorrectionType = UITextAutocorrectionTypeNo;
            self.textInput.autocapitalizationType = UITextAutocapitalizationTypeNone;
            break;
```

Not sure about android

**Comment by minggo** [comment]
Android doesn't have the issue.

**Comment by minggo** [comment]
@jasonchin Will you please send a pull request?

**Comment by jasonchin** [comment]
@minggo Sorry, I don't have git or the repository setup so I can't. Thought I would just confirm the bug and share the fix.

**Comment by minggo** [comment]
@jasonchin Got it, i will send a PR to fix it.

**Comment by minggo** [comment]
Fixed in #15788

**Comment by rynkowsg** [comment]
Originally I was referring to the problem that "EditBox by default uses capital letters" for all characters. And the problem is common for both Android and iOS (INITIAL_CAPS_ALL_CHARACTERS is default flag).

As I wrote I used SENSITIVE flag as a partly worked workaround. @minggo, in #15788 you improved that workaround - for SENSITIVE flag all characters are small. But SENSITIVE should focus on being sensitive only:

```
             /**
             * Indicates that the text entered is sensitive data that the
             * implementation must never store into a dictionary or table for use
             * in predictive, auto-completing, or other accelerated input schemes.
             * A credit card number is an example of sensitive data.
             */
            SENSITIVE,
```

Would it be better to:
- add an additional flag, something like LOWERCASE_ALL_CARACTERS and
- implement passing couple flags together (using bitwise inclusive OR), sample:

```
_emailEditBox->setInputFlag( InputFlag::SENSITIVE | InputFlag::LOWERCASE_ALL_CARACTERS );
```

?

**Comment by minggo** [comment]
@rynkowsg Sorry for my misunderstood. But @zilongshanren told me that he tested on Android and didn't have the problem.

> add an additional flag, something like LOWERCASE_ALL_CARACTERS

I think it is reasonable, and i think UPPERCASE_ALL_CARACTERS is needed too, because you may want to change to upper case after you set lower case.

**Comment by rynkowsg** [comment]
Uppercase is there already:

```
            /**
             * Capitalize all characters automatically.
             */
            INITIAL_CAPS_ALL_CHARACTERS,
```

and as I mentioned before this is a default flag for EditBoxes.

**Comment by minggo** [comment]
Yep, you are right.
I have to discuss with @zilongshanren about the feature.

And the issue is
- Need a way such as LOWERCASE_ALL_CARACTERS
- and use LOWERCASE_ALL_CARACTERS by default

right?

**Comment by rynkowsg** [comment]
Yep, exactly.

**Comment by rynkowsg** [comment]
Just one thing, LOWERCASE_ALL_CHARACTERS not LOWERCASE_ALL_CARACTERS. I did a typo  in my previous comment. Sorry.

**Comment by minggo** [comment]
fixed in #15797
